packages/components/src/range-control/stories/index.js

```diff
@@ -1,145 +1,127 @@
+		},
+		allowReset: { control: { type: 'boolean' } },
+		beforeIcon: {
+			control: { type: 'select' },
+			options: Object.keys( ICONS ),
+			mapping: ICONS,
+		},
+		color: { control: { type: 'color' } },
+		disabled: { control: { type: 'boolean' } },
+		help: { control: { type: 'text' } },
+		marks: { control: { type: 'object' } },
+		min: { control: { type: 'number' } },
+		max: { control: { type: 'number' } },
+		railColor: { control: { type: 'color' } },
+		showTooltip: { control: { type: 'boolean' } },
+		step: { control: { type: 'number' } },
+		trackColor: { control: { type: 'color' } },
+		withInputField: { control: { type: 'boolean' } },
+	},
 	parameters: {
-		knobs: { disable: false },
+		docs: { source: { state: 'open' } },
 	},
 };
 
-const RangeControlWithState = ( props ) => {
-	const initialValue = props.value === undefined ? 5 : props.value;
+const RangeControlWithState = ( { initialValue, ...props } ) => {
 	const [ value, setValue ] = useState( initialValue );
 
 	return <RangeControl { ...props } value={ value } onChange={ setValue } />;
 };
 
-const DefaultExample = () => {
-	const [ value, setValue ] = useState( undefined );
-
-	const showBeforeIcon = boolean( 'beforeIcon', false );
-	const showAfterIcon = boolean( 'afterIcon', false );
-
-	const props = {
-		afterIcon: showAfterIcon ? wordpress : undefined,
-		allowReset: boolean( 'allowReset', false ),
-		beforeIcon: showBeforeIcon ? wordpress : undefined,
-		color: text( 'color', COLORS.ui.theme ),
-		disabled: boolean( 'disabled', false ),
-		help: text( 'help', '' ),
-		label: text( 'label', 'Range Label' ),
-		marks: boolean( 'marks', false ),
-		max: number( 'max', 100 ),
-		min: number( 'min', 0 ),
-		showTooltip: boolean( 'showTooltip', false ),
-		step: text( 'step', 1 ),
-		railColor: text( 'railColor', null ),
-		trackColor: text( 'trackColor', null ),
-		withInputField: boolean( 'withInputField', true ),
-		value,
-		onChange: setValue,
-	};
-
-	return (
-		<Wrapper>
-			<RangeControl { ...props } />
-		</Wrapper>
-	);
-};
-
-const RangeControlLabeledByMarksType = ( props ) => {
-	const label = Array.isArray( props.marks ) ? 'Custom' : 'Automatic';
-	return <RangeControl { ...{ ...props, label } } />;
+export const Default = RangeControlWithState.bind( {} );
+Default.args = {
+	label: 'Range label',
 };
 
-export const _default = () => {
-	return <DefaultExample />;
+/**
+ * The `initialPosition` prop sets the starting position of the slider when no `value` is provided.
+ */
+export const InitialValueZero = RangeControlWithState.bind( {} );
+InitialValueZero.args = {
+	...Default.args,
+	initialPosition: 0,
+	max: 20,
 };
 
-export const InitialValueZero = () => {
-	const label = text( 'Label', 'How many columns should this use?' );
+/**
+ * Setting the `step` prop to `"any"` will allow users to select non-integer values.
+ * This also overrides both `withInputField` and `showTooltip` props to `false`.
+ */
+export const WithAnyStep = ( props ) => {
```

Comment:
Should we use `RangeControlWithState` instead? (in that case, the additional `initialValue` prop would actually come in handy)

Reply:
I kind of thought it was useful to have the current value showing for this example, otherwise it's impossible to see that it's a float.

Reply:
That is a good point 😅  Let's leave it as-is